### ISSUE TITLE
[self-update] command shouldn't require running in a Drupal docroot

### ISSUE THREAD

**Issue by geerlingguy** [hechoendrupal/DrupalConsole#615]
It seems the `self-update` command only works if you run the command within a valid Drupal site docroot. If you run the command elsewhere, you get:

```
  [InvalidArgumentException]             
  Command "self-update" is not defined.  
```

Shouldn't it be possible to update Drupal Console independent of a Drupal installation/codebase? I'm asking specifically in reference to my Ansible role that installs Drupal Console and otherwise works perfectly without referencing any particular Drupal installation on the server (which is nice for scaffolding the server before installation/deploying Drupal itself); see https://github.com/geerlingguy/ansible-role-drupal-console/issues/1 .

**Comment by jmolivas** [comment]
@geerlingguy I did some work on how commands are registered on this PR https://github.com/hechoendrupal/DrupalConsole/pull/656 

At this point we have two commands that shouldn't require running in a Drupal docroot `init` and `self-update`

* This is how it looks when console is executed outside of a drupal root
![screen shot 2015-06-07 at 8 43 42 pm](https://cloud.githubusercontent.com/assets/366275/8029918/2152a0b2-0d77-11e5-80c3-282809d9ebcc.png)

* This is how it looks when console is executed in a drupal root, but drupal not installed yet
![screen shot 2015-06-07 at 8 43 56 pm](https://cloud.githubusercontent.com/assets/366275/8029928/355f1bd0-0d77-11e5-88f8-4387ac97a7e8.png)

**Comment by jmolivas** [comment]
This is how it looks to execute the `self-update` command outside of a Drupal docroot  

![screen shot 2015-06-08 at 1 20 00 am](https://cloud.githubusercontent.com/assets/366275/8030590/efac22bc-0d7c-11e5-9b31-7d0204d0bf7b.png)
Drupal docroot

**Comment by jmolivas** [comment]
This is how it looks to execute the `init` command outside of a Drupal docroot

![screen shot 2015-06-08 at 2 03 41 am](https://cloud.githubusercontent.com/assets/366275/8031377/a8b2a04c-0d82-11e5-8a57-60dbc692d025.png)

**Comment by jmolivas** [comment]
@geerlingguy this was merged and it was included on latest release [0.7.14](https://github.com/hechoendrupal/DrupalConsole/releases/tag/0.7.14)

**Comment by geerlingguy** [comment]
@jmolivas - Thanks! I've updated my Ansible role to allow self-update to work any time.
